tensorflow_addons/layers/noisy_dense.py

```diff
@@ -0,0 +1,274 @@
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+import tensorflow as tf
+from tensorflow.keras import (
+    activations,
+    initializers,
+    regularizers,
+    constraints,
+)
+from tensorflow.keras import backend as K
+from tensorflow.keras.layers import InputSpec
+
+
+@tf.keras.utils.register_keras_serializable(package="Addons")
+class NoisyDense(tf.keras.layers.Layer):
+    """
+    Like normal dense layer (https://github.com/tensorflow/tensorflow/blob/v2.3.0/tensorflow/python/keras/layers/core.py#L1067-L1233)
+    but random noisy is added to the weights matrix. But as the network improves the random noise is decayed until it is insignificant.
+
+    A `NoisyDense` layer implements the operation:
+    `output = activation(dot(input, µ_kernel + (σ_kernel * ε_kernel)) + bias)`
```

Comment:
Please get rid of unicode symbol, you can use mathjax expr for rendering math expression.
```
$$
\mathrm{NoisyDense}(x) = \mathrm{activation}(\mathrm{dot}(x, \mu + (\sigma \cdot \epsilon)) + \mathrm{bias})
$$
```

Reply:
Yes I think we need to add this somwhere contribution guidelines.

Reply:
How do you want me to add mathjax expressions to the comments? Or should I just use a raw string and leave it unrendered?

Reply:
Generally it was suggested for Markdown files:
https://github.com/tensorflow/docs/blob/master/site/en/community/contribute/docs_style.md#math-in-markdown
But:
> MathJax does not render properly on GitHub.
This notation can be off-putting to unfamiliar developers.

@WindQAQ It would be nice to have something like http://asciimath.org/ to have both source code and rendered MathJax readability

Reply:
Either way in docstring is fine to me. For real python code, please use `mu` instead of `µ` for example.

Reply:
Yes `mu` is also the asciimath style the then couldbe rendered as a MathJax /cc @lamberta just to know if he have already explored this option.